x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/MlSingleNodeTestCase.java

```diff
@@ -128,11 +149,66 @@ protected void waitForMlTemplates() throws Exception {
         return responseHolder.get();
     }
 
+    protected static ThreadPool mockThreadPool() {
+        ThreadPool tp = mock(ThreadPool.class);
+        ExecutorService executor = mock(ExecutorService.class);
+        doAnswer(invocationOnMock -> {
+            ((Runnable) invocationOnMock.getArguments()[0]).run();
+            return null;
+        }).when(executor).execute(any(Runnable.class));
+        when(tp.executor(any(String.class))).thenReturn(executor);
+        doAnswer(invocationOnMock -> {
+            ((Runnable) invocationOnMock.getArguments()[0]).run();
+            return null;
+        }).when(tp).schedule(
+            any(Runnable.class), any(TimeValue.class), any(String.class)
+        );
+        return tp;
+    }
+
+
     public static void assertNoException(AtomicReference<Exception> error) throws Exception {
         if (error.get() == null) {
             return;
         }
         throw error.get();
     }
 
+    public static class MockPainlessScriptEngine extends MockScriptEngine {
```

Comment:
I don't understand why this is now required. Does it have to be part of this change?

Reply:
I noticed log spam when it wasn't around. This and the other changes (including more plugins) got rid of a bunch of log spam for the single node tests.